app/controllers/user_mentee_application_cohorts_controller.rb

```diff
@@ -8,7 +8,24 @@ def index
 
   def show
     @cohort = UserMenteeApplicationCohort
-              .includes(:user_mentee_applications)
+              .includes(user_mentee_applications: :mentee_application_states)
               .find(params[:id])
+
+    @filtered_applications = filter_applications(@cohort.user_mentee_applications)
+  end
+
+  private
+
+  def filter_applications(applications)
```

Comment:
This is fine for now. I'd need to pull it down locally to see but I am pretty sure we can do some ActiveRecord joins on scopes to optimize this. Not worth the trouble for now though.

Reply:
I'd definitely like to give it a try in a future Issue.